core/src/main/java/de/danoeh/antennapod/core/sync/SynchronizationSettings.java

```diff
@@ -0,0 +1,100 @@
+                .putLong(SynchronizationSharedPreferenceKeys.LAST_SYNC_ATTEMPT_TIMESTAMP, 0)
+                .apply();
+    }
+
+    public static boolean isLastSyncSuccessful() {
+        return getSharedPreferences().getBoolean(SynchronizationSharedPreferenceKeys.LAST_SYNC_ATTEMPT_SUCCESS, false);
+    }
+
+    public static long getLastSyncAttempt() {
+        return getSharedPreferences().getLong(SynchronizationSharedPreferenceKeys.LAST_SYNC_ATTEMPT_TIMESTAMP, 0);
+    }
+
+    public static String getSelectedSyncProviderKey() {
+        return getSharedPreferences().getString(SynchronizationSharedPreferenceKeys.SELECTED_SYNC_PROVIDER, null);
+    }
+
+    public static void setIsProviderConnected(boolean isConnected) {
+        getSharedPreferences()
+                .edit()
+                .putBoolean(SynchronizationSharedPreferenceKeys.IS_SYNC_PROVIDER_CONNECTED, isConnected)
+                .apply();
+    }
+
+    public static boolean isProviderConnected() {
+        return getSharedPreferences()
+                .getBoolean(SynchronizationSharedPreferenceKeys.IS_SYNC_PROVIDER_CONNECTED, false);
+    }
+
+    public static void setSelectedSyncProvider(String userSelect) {
+        getSharedPreferences()
+                .edit()
+                .putString(SynchronizationSharedPreferenceKeys.SELECTED_SYNC_PROVIDER, userSelect)
+                .apply();
+    }
+
+    public static void updateLastSynchronizationAttempt() {
+        getSharedPreferences().edit()
+                .putLong(SynchronizationSharedPreferenceKeys.LAST_SYNC_ATTEMPT_TIMESTAMP, System.currentTimeMillis())
+                .apply();
+    }
+
+    public static void setLastSynchronizationAttemptSuccess(boolean isSuccess) {
+        getSharedPreferences().edit()
+                .putBoolean(SynchronizationSharedPreferenceKeys.LAST_SYNC_ATTEMPT_SUCCESS, isSuccess)
+                .apply();
+    }
+
+    public static long getLastSubscriptionSynchronizationTimestamp() {
+        return getSharedPreferences().getLong(SynchronizationSharedPreferenceKeys.LAST_SUBSCRIPTION_SYNC_TIMESTAMP, 0);
+    }
+
+    public static void clearFeedQueues() {
+        getSharedPreferences().edit()
+                .putString(SynchronizationSharedPreferenceKeys.QUEUED_FEEDS_ADDED, "[]")
+                .putString(SynchronizationSharedPreferenceKeys.QUEUED_FEEDS_REMOVED, "[]")
+                .apply();
+    }
+
+    public static void setLastSubscriptionSynchronizationAttemptTimestamp(long newTimeStamp) {
+        getSharedPreferences().edit()
+                .putLong(SynchronizationSharedPreferenceKeys.LAST_SUBSCRIPTION_SYNC_TIMESTAMP, newTimeStamp).apply();
+    }
+
+    public static long getLastEpisodeActionSynchronizationTimestamp() {
+        return getSharedPreferences()
+                .getLong(SynchronizationSharedPreferenceKeys.LAST_EPISODE_ACTIONS_SYNC_TIMESTAMP, 0);
+    }
+
+    public static void clearEpisodeActionQueue() {
```

Comment:
Same here, I think this fits better into `SynchronizationQueue`

Reply:
🤦